### PR TITLE
add multi-windows support

### PR DESCRIPTION
**Pull request by freeboy**
Re: #

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by freeboy** [comment]
* Replace AppActivity.java in your project path "[project name]/native/engine/android/app/src/com/cocos/game/AppActivity.java"
* [AppActivity.zip](https://github.com/cocos/cocos-engine/files/8889735/AppActivity.zip)

File: native/cocos/bindings/auto/jsb_gfx_auto.cpp [hunk 1]
```diff
@@ -21418,6 +21418,9 @@ SE_BIND_FUNC(js_gfx_Device_createShader)
 
 static bool js_gfx_Device_createSwapchain(se::State& s) // NOLINT(readability-identifier-naming)
 {
+	// linwei: multi-windows hack
+    return true;
+
```
Comment (star-e) [inline native/cocos/bindings/auto/jsb_gfx_auto.cpp:22722]: native/cocos/bindings/auto/XXX is auto generated. Modifications might be overwritten.

File: native/cocos/application/BaseGame.cpp [hunk 2]
```diff
@@ -41,12 +42,18 @@ int BaseGame::init() {
...
+        //if (_windowInfo.x == -1 || _windowInfo.y == -1) {
+        //    createWindow(_windowInfo.title.c_str(), _windowInfo.width, _windowInfo.height, _windowInfo.flags);
+        //} else {
+        //    createWindow(_windowInfo.title.c_str(),
+        //                 _windowInfo.x, _windowInfo.y, _windowInfo.width, _windowInfo.height, _windowInfo.flags);
+        //}
+
+        ISystemWindowManager *windowMgr = BasePlatform::getPlatform()->getInterface<ISystemWindowManager>();
+        ISystemWindow* window1 = windowMgr->createWindow("w1");
+        window1->createWindow(_windowInfo.title.c_str(), _windowInfo.width, _windowInfo.height, _windowInfo.flags);
```
Comment (minggo) [inline native/cocos/application/BaseGame.cpp:59]: Has both `ISystemWindowManager::createWindow()` and `ISystemWindow::createWindow()` is strange.

File: native/cocos/core/Root.cpp [hunk 3]
```diff
@@ -67,28 +70,43 @@ Root::~Root() {
...
+        info.width  = size.x;
+        info.height = size.y;
+        info.windowHandle = (void *)handle;
 
-    auto &depthStencilAttachment = renderPassInfo.depthStencilAttachment;
-    depthStencilAttachment.format = swapchain->getDepthStencilTexture()->getFormat();
-    depthStencilAttachment.depthStoreOp = gfx::StoreOp::DISCARD;
-    depthStencilAttachment.stencilStoreOp = gfx::StoreOp::DISCARD;
+        gfx::Swapchain *swapchain = gfx::Device::getInstance()->createSwapchain(info);
+        _swapchains.emplace_back(swapchain);
```
Comment (minggo) [inline native/cocos/core/Root.cpp:90]: Why create and hold swapchains in root?

File: native/cocos/core/Root.cpp [hunk 4]
```diff
@@ -292,9 +310,13 @@ void Root::frameMove(float deltaTime, int32_t totalFrames) {
...
 
     if (_pipelineRuntime != nullptr && !_cameraList.empty()) {
-        _swapchains.clear();
-        _swapchains.emplace_back(_swapchain);
+        //_swapchains.clear();
+        //_swapchains.emplace_back(_swapchain);
         _device->acquire(_swapchains);
+
+        _cameraList[0]->changeTargetWindow(_windows[0]);
+        _cameraList[1]->changeTargetWindow(_windows[1]);
```
Comment (minggo) [inline native/cocos/core/Root.cpp:336]: Camera bind window every frame? And how to know bind which window?

File: native/cocos/application/BaseGame.cpp (hunk 2 above)
Comment (freeboy) [reply native/cocos/application/BaseGame.cpp:59]: maybe ISystemWindow::create() ?

File: native/cocos/core/Root.cpp (hunk 4 above)
Comment (freeboy) [reply native/cocos/core/Root.cpp:336]: Just for testing

File: native/cocos/bindings/auto/jsb_gfx_auto.cpp (hunk 1 above)
Comment (freeboy) [reply native/cocos/bindings/auto/jsb_gfx_auto.cpp:22722]: You're right，just a testing code.

File: native/cocos/core/Root.cpp (hunk 4 above)
Comment (freeboy) [reply native/cocos/core/Root.cpp:336]: No need binging every frame，the target window can be specified from Editor.

File: native/cocos/application/BaseGame.cpp (hunk 2 above)
Comment (minggo) [reply native/cocos/application/BaseGame.cpp:59]: Why just use `ISystemWindowManager::createWindow()` to create a real window?

File: native/cocos/core/Root.cpp (hunk 4 above)
Comment (minggo) [reply native/cocos/core/Root.cpp:336]: Got it, then it is better to add comments to say `testing codes`.

File: native/cocos/application/BaseGame.cpp [hunk 5]
```diff
@@ -41,12 +42,18 @@ int BaseGame::init() {
...
-                         _windowInfo.x, _windowInfo.y, _windowInfo.width, _windowInfo.height, _windowInfo.flags);
-        }
+        //if (_windowInfo.x == -1 || _windowInfo.y == -1) {
+        //    createWindow(_windowInfo.title.c_str(), _windowInfo.width, _windowInfo.height, _windowInfo.flags);
+        //} else {
+        //    createWindow(_windowInfo.title.c_str(),
+        //                 _windowInfo.x, _windowInfo.y, _windowInfo.width, _windowInfo.height, _windowInfo.flags);
+        //}
+
+        ISystemWindowManager *windowMgr = BasePlatform::getPlatform()->getInterface<ISystemWindowManager>();
```
Comment (qiuguohua) [inline native/cocos/application/BaseGame.cpp:57]: It is recommended to use the CC_GET_PLATFORM_INTERFACE macro

File: native/cocos/platform/SDLHelper.cpp [hunk 6]
```diff
@@ -146,6 +146,7 @@ int windowFlagsToSDLWindowFlag(int flags) {
 namespace cc {
 SDLHelper::SDLHelper(IEventDispatch *delegate)
 : _delegate(delegate) {
+    _instance = this;
```
Comment (qiuguohua) [inline native/cocos/platform/SDLHelper.cpp:149]: Is it necessary to reset to nullptr in the destructor？

File: native/cocos/platform/SDLHelper.cpp [hunk 7]
```diff
@@ -335,44 +342,50 @@ void SDLHelper::pollEvent(bool *quit) {
...
 }
 
-bool SDLHelper::createWindow(const char *title,
+SDL_Window *SDLHelper::createWindow(const char *title,
                              int x, int y, int w,
                              int h, int flags) {
-    if (_isWindowCreated) {
-        return true;
-    }
+    //if (_isWindowCreated) {
```
Comment (qiuguohua) [inline native/cocos/platform/SDLHelper.cpp:362]: If you comment out here, you need to pay attention to the platforms of linux and QNX, which are common to multiple platforms

File: native/cocos/application/BaseGame.cpp (hunk 5 above)
Comment (freeboy) [reply native/cocos/application/BaseGame.cpp:57]: OK，thanks.

File: native/cocos/core/Root.cpp (hunk 4 above)
Comment (freeboy) [reply native/cocos/core/Root.cpp:336]: 👌

**Comment by freeboy** [comment]
https://github.com/cocos/cocos-engine/pull/12243 instead

**Pull request closed without merging**